lib/pantry/handlers/alexa_request_handler.rb

```diff
@@ -4,18 +4,27 @@ module Handlers
 
     class AlexaRequestHandler
 
-      attr_reader :request, :session
+      attr_reader :request, :user_id
       def initialize(params)
-        @request = AlexaRubykit.build_request(params)
+        @request = AlexaRubykit.build_request params
+        @user_id = determine_user request
       end
 
       def process
-        handler = handler_for[request_type].new
-        handler.process session: request.session
+        handler = handler_for(request.type).new
+        handler.process request: request, user_id: user_id
       end
 
       private
 
+      def determine_user(request)
+        current_user = User.find_by amazon_user_id: request.session.user['userId']
+        if !current_user.present?
+          current_user = User.create amazon_user_id: request.session.user['userId']
+        end
+        current_user.id
+      end
```

Comment:
I believe Rails has something like `User.find_or_create_by` that you could use for this.